Source/WTF/wtf/text/StringView.cpp

```diff
@@ -318,26 +319,26 @@ AtomString StringView::convertToASCIILowercaseAtom() const
 WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
 // FIXME: Take in a span.
 template<typename DestinationCharacterType, typename SourceCharacterType>
-void getCharactersWithASCIICaseInternal(StringView::CaseConvertType type, DestinationCharacterType* destination, std::span<const SourceCharacterType> source)
+void getCharactersWithASCIICaseInternal(StringView::CaseConvertType type, std::span<DestinationCharacterType> destination, std::span<const SourceCharacterType> source)
 {
     static_assert(std::is_same<SourceCharacterType, LChar>::value || std::is_same<SourceCharacterType, UChar>::value);
     static_assert(std::is_same<DestinationCharacterType, LChar>::value || std::is_same<DestinationCharacterType, UChar>::value);
     static_assert(sizeof(DestinationCharacterType) >= sizeof(SourceCharacterType));
     auto caseConvert = (type == StringView::CaseConvertType::Lower) ? toASCIILower<SourceCharacterType> : toASCIIUpper<SourceCharacterType>;
-    for (auto character : source)
-        *destination++ = caseConvert(character);
+    for (auto [i, character] : IndexedRange(source))
+        destination[i] = caseConvert(character);
 }
 WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
 
 // FIXME: Take in a span.
-void StringView::getCharactersWithASCIICase(CaseConvertType type, LChar* destination) const
+void StringView::getCharactersWithASCIICase(CaseConvertType type, std::span<LChar> destination) const
 {
```

Comment:
I guess we can remove the FIXME?

Reply:
Yes, I've removed the FIXMEs and the WTF_ALLOW_UNSAFE_BUFFER_USAGE.